src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -52,15 +52,19 @@ Copyright (c) .NET Foundation. All rights reserved.
           work without user intervention. The current static evaluation
           requirement limits us.
    -->
-  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and 
-                            '$(HasRuntimeOutput)' == 'true' and 
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and
+                            '$(HasRuntimeOutput)' == 'true' and
                             '$(OS)' == 'Windows_NT' and
                             '$(RuntimeIdentifier)' == ''">
     <_UsingDefaultRuntimeIdentifier>true</_UsingDefaultRuntimeIdentifier>
     <RuntimeIdentifier Condition="'$(PlatformTarget)' == 'x64'">win7-x64</RuntimeIdentifier>
     <RuntimeIdentifier Condition="'$(PlatformTarget)' == 'x86' or '$(PlatformTarget)' == ''">win7-x86</RuntimeIdentifier>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(UseCurrentRuntimeIdentifier)' == 'true'">
+    <RuntimeIdentifier>$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>
```

Comment:
@am11 @dsplaisted @eerhardt 

On a Microsoft SDK, `UseCurrentRuntimeIdentifier` is the same as portable rid. So `--use-current-runtime` means publish portable.
On a source-build SDK, this is the same as the machine rid. So the result may be a more specific publish that doesn't work portable.

I think this difference will bite source-build SDK users, and it is preferable to have a flag that means portable-rid consistently.

Reply:
I do not know the full background why source-build chose to override it, but.. if the value of `NETCoreSdkRuntimeIdentifier` is overridden in source-build repo for set of reasons, do those same reasons not apply to `RuntimeIdentifier`?

In other words, `--use-current-runtime`'s docs are stating "Use current runtime as the target runtime.", the one SDK is executing on. This is what we have implemented. It is not strictly calling out `portable-rid`, that's an internal detail of general (non-overridden, non-source-build) case.

Reply:
> I do not know the full background why source-build chose to override it, but.. if the value of NETCoreSdkRuntimeIdentifier is overridden in source-build repo for set of reasons, do those same reasons not apply to RuntimeIdentifier?

source-build does it to indicate the resulting artifacts are not portable.

> In other words, --use-current-runtime's docs are stating "Use current runtime as the target runtime.", the one SDK is executing on. This is what we have implemented. It is not strictly calling out portable-rid, that's an internal detail of general (non-overridden, non-source-build) case.

The main use-case is to enable easy publish for the portable rid.
If this flag enables that on Microsoft sdk, that is what it will be used for.
So I'd like this to have the same behavior on source-build sdk, and we can update the docs/implementation accordingly.

Reply:
> except maybe the word "machine"

It was adapted from @eerhardt's question https://github.com/dotnet/sdk/pull/14093#issuecomment-708365676 (and maybe the fact 'machine config' is a thing in .NET Framework) 😉